0.36.0 build failure with undefined symbols

### What's wrong?

```
# github.com/pyroscope-io/godeltaprof/internal/pprof
../pkg/mod/github.com/pyroscope-io/godeltaprof@v0.1.1/internal/pprof/delta_mutex.go:25:20: undefined: runtime_cyclesPerSecond
../pkg/mod/github.com/pyroscope-io/godeltaprof@v0.1.1/internal/pprof/proto.go:400:8: undefined: runtime_expandFinalInlineFrame
```

### Steps to reproduce

`go build cmd/grafana-agent`

### System information

ArchLinux

### Software version

v0.36.0

### Configuration

_No response_

### Logs

_No response_

--- a/go.mod
+++ b/go.mod
@@ -57,6 +57,7 @@ require (
 	github.com/grafana/dskit v0.0.0-20230725162534-96da0816983e
 	github.com/grafana/go-gelf/v2 v2.0.1
 	github.com/grafana/loki v1.6.2-0.20230730074919-e455a11b49dd // k161 branch
+	github.com/grafana/pyroscope-go/godeltaprof v0.1.3
 	github.com/grafana/pyroscope/api v0.2.0
 	github.com/grafana/pyroscope/ebpf v0.2.1
 	github.com/grafana/regexp v0.0.0-20221123153739-15dc172cd2db
@@ -150,7 +151,6 @@ require (
 	github.com/prometheus/prometheus v1.99.0
 	github.com/prometheus/snmp_exporter v0.22.1-0.20230623130038-562ae9055ce3
 	github.com/prometheus/statsd_exporter v0.22.8
-	github.com/pyroscope-io/godeltaprof v0.1.2
 	github.com/richardartoul/molecule v1.0.1-0.20221107223329-32cfee06a052
 	github.com/rs/cors v1.9.0
 	github.com/shirou/gopsutil/v3 v3.23.5

--- a/pkg/server/server.go
+++ b/pkg/server/server.go
@@ -17,6 +17,7 @@ import (
 	"github.com/go-kit/log/level"
 	"github.com/gorilla/mux"
 	"github.com/grafana/ckit/memconn"
+	_ "github.com/grafana/pyroscope-go/godeltaprof/http/pprof" // anonymous import to get the godeltaprof handler registered
 	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
 	"github.com/hashicorp/go-multierror"
 	"github.com/oklog/run"
@@ -24,7 +25,6 @@ import (
 	"github.com/opentracing/opentracing-go"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
-	_ "github.com/pyroscope-io/godeltaprof/http/pprof" // anonymous import to get the godeltaprof handler registered
 	"github.com/weaveworks/common/logging"
 	"github.com/weaveworks/common/middleware"
 	"golang.org/x/net/netutil"

--- a/service/http/http.go
+++ b/service/http/http.go
@@ -7,6 +7,7 @@ import (
 	"fmt"
 	"net"
 	"net/http"
+	_ "net/http/pprof" // Register pprof handlers
 	"path"
 	"sort"
 	"strings"
@@ -18,6 +19,7 @@ import (
 	"github.com/grafana/agent/component"
 	"github.com/grafana/agent/service"
 	"github.com/grafana/ckit/memconn"
+	_ "github.com/grafana/pyroscope-go/godeltaprof/http/pprof" // Register godeltaprof handler
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"go.opentelemetry.io/contrib/instrumentation/github.com/gorilla/mux/otelmux"

